### PR TITLE
Remove the additional host register calls initially intended for performance improvement on Grace Hopper

### PR DESCRIPTION
**Pull request by kingcrimsontianyu**
## Description
On Grace Hopper, file I/O takes a special path that calls `cudaHostRegister` to circumvent a performance issue. Recent benchmark shows that this workaround is no longer necessary . This PR is making a clean-up.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/io/utilities/datasource.cpp [hunk 1]
```diff
@@ -134,27 +134,6 @@ class file_source : public datasource {
...
-    result_cache[deviceId] = (props.pageableMemoryAccessUsesHostPageTables == 1);
-    CUDF_LOG_INFO(
-      "Device {} pageableMemoryAccessUsesHostPageTables: {}", deviceId, result_cache[deviceId]);
-  }
-
-  return result_cache[deviceId];
-}
-
 /**
  * @brief Implementation class for reading from a file using memory mapped access.
```
Comment (vuule) [inline cpp/src/io/utilities/datasource.cpp:138]: We can now remove the `min_size_estimate` parameter, but I'm okay with keeping it. Let me know what you think. If we do keep it, we should probably mark it as `[[maybe_unused]]`.

Comment (kingcrimsontianyu) [reply cpp/src/io/utilities/datasource.cpp:138]: Thanks for capturing this. I've chosen to remove it.
It's a bit surprising that either the compiler (isn't there `-Wunused` ?) or linter (is there any?) didn't complain about this unused parameter. I need to have a better understanding of our build system.

File: cpp/src/io/functions.cpp [hunk 2]
```diff
@@ -123,15 +123,13 @@ namespace {
 
 std::vector<std::unique_ptr<cudf::io::datasource>> make_datasources(source_info const& info,
                                                                     size_t offset            = 0,
-                                                                    size_t max_size_estimate = 0,
-                                                                    size_t min_size_estimate = 0)
+                                                                    size_t max_size_estimate = 0)
```
Comment (kingcrimsontianyu) [inline cpp/src/io/functions.cpp:126]: Removing `min_size_estimate` should make this PR have a `breaking` label. Is that correct? @vuule

File: cpp/include/cudf/io/datasource.hpp [hunk 3]
```diff
@@ -86,28 +86,24 @@ class datasource {
...
    * @brief Creates a source from a file path.
    *
-   * @note Parameters `offset`, `max_size_estimate` and `min_size_estimate` are hints to the
-   * `datasource` implementation about the expected range of the data that will be read. The
-   * implementation may use these hints to optimize the read operation. These parameters are usually
-   * based on the byte range option. In this case, `min_size_estimate` should be no greater than the
-   * byte range to avoid potential issues when reading adjacent ranges. `max_size_estimate` can
-   * include padding after the byte range, to include additional data that may be needed for
-   * processing.
+   * @note Parameters `offset` and `max_size_estimate` are hints to the `datasource`
```
Comment (davidwendt) [inline cpp/include/cudf/io/datasource.hpp:89]:
```suggestion
   * Parameters `offset` and `max_size_estimate` are hints to the `datasource`
```
I don't think `@note` is necessary here.

File: cpp/include/cudf/io/datasource.hpp [hunk 4]
```diff
@@ -86,28 +86,24 @@ class datasource {
...
+   * implementation about the expected range of the data that will be read. The
+   * implementation may use these hints to optimize the read operation. These parameters
+   * are usually based on the byte range option. In this case, `max_size_estimate` can
+   * include padding after the byte range, to include additional data that may be needed
+   * for processing.
    *
    @throws cudf::logic_error if the minimum size estimate is greater than the maximum size estimate
```
Comment (davidwendt) [inline cpp/include/cudf/io/datasource.hpp:96]:
```suggestion
   * @throws cudf::logic_error if the minimum size estimate is greater than the maximum size estimate
```

File: cpp/include/cudf/io/datasource.hpp (hunk 3 above)
Comment (kingcrimsontianyu) [reply cpp/include/cudf/io/datasource.hpp]: Thanks! Done. 
Also the comment section has been reformatted.

File: cpp/include/cudf/io/datasource.hpp (hunk 4 above)
Comment (kingcrimsontianyu) [reply cpp/include/cudf/io/datasource.hpp]: Completed.

File: cpp/include/cudf/io/datasource.hpp [hunk 5]
```diff
@@ -86,28 +86,24 @@ class datasource {
...
+   * Parameters `offset` and `max_size_estimate` are hints to the `datasource` implementation about
+   * the expected range of the data that will be read. The implementation may use these hints to
+   * optimize the read operation. These parameters are usually based on the byte range option. In
+   * this case, `max_size_estimate` can include padding after the byte range, to include additional
+   * data that may be needed for processing.
    *
-   @throws cudf::logic_error if the minimum size estimate is greater than the maximum size estimate
+   * @throws cudf::logic_error if the minimum size estimate is greater than the maximum size
+   * estimate
    *
```
Comment (vuule) [inline cpp/include/cudf/io/datasource.hpp:97]: this can just be removed
```suggestion
```

File: cpp/src/io/functions.cpp (hunk 2 above)
Comment (vuule) [reply cpp/src/io/functions.cpp:126]: That's correct, even though the breakage doesn't affect any users.

File: cpp/include/cudf/io/datasource.hpp (hunk 5 above)
Comment (kingcrimsontianyu) [reply cpp/include/cudf/io/datasource.hpp]: Thanks! Done.

**Review by vuule — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

**Pull request merged**